tools/serverpod_cli/lib/src/language_server/language_server.dart

```diff
@@ -88,6 +180,19 @@ List<Diagnostic> _validateYamlProtocol(String yaml, String sourcePath) {
       ),
     );
   }).toList();
+}
+
+bool _isProtocolInServerPath(Uri uri, Uri? serverUri) {
```

Comment:
Suggestion: Make serverUri non nullable.

Seems unnecessary to call this method if we don't know what the server path is. Maybe it makes more sense to handle that before we call the method?

Reply:
postponing this as per our discussion.